astropy/io/fits/convenience.py

```diff
@@ -463,6 +462,8 @@ def table_to_hdu(table):
 
     # Not all tables with mixin columns are supported
     if table.has_mixin_columns:
+        from ...table.column import BaseColumn
```

Comment:
Add a comment note why this isn't at the top of the file, otherwise it may be tried to to moved around later in a PEP8 cleanup.